src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SnakeRoadCheck.java

```diff
@@ -160,6 +161,10 @@ private boolean networkWalkQualifiesAsSnakeRoad(final SnakeRoadNetworkWalk walk)
      */
     private void walkNetwork(final Edge current, final SnakeRoadNetworkWalk walk)
     {
+        final Set<Edge> directConnections = walk.getConnectedMasterEdgeOfTheSameWay(current);
+        directConnections.stream().forEach(connection -> walk
```

Comment:
Isn't this change performing the same operation as before? For all connections, ```checkIfEdgeHeadingDifferenceExceedsThreshold(comingFrom, comingTo);``` was called from walk.visitedEdge previously, and now it is called here. Was this to improve performance or is there something I am missing?

Reply:
The difference is before it used to call checkIfEdgeHeadingDifferenceExceedsThreshold() from visitedEdge() for all the edges not just directly connected edges to a current edge. My code will make sure that it will check the heading difference of two directly connected edges to a current edge.

Reply:
ahh I think the names got me confused.. directConnections and walk.getDirectConnections() are different but I read it as same.

Reply:
The requirements from your PR description say to me that the 'snakiness' can be determined from any 2 consecutive edges, so shouldn't we be calling lines 164-166 inside the while loop as well? Instead of passing ```current``` we pass ```connection```

Reply:
Heading difference is verified only against the current edge and it direct connections(this will be done for each edge when it is processed). Valence is checked for whole highway so that we can qualify it as a snake road.
Moving these lines inside the while loop will make the check do the same thing repeatedly for an edge object that will be processed for snakeRoadCheck.

Reply:
Sorry if I'm misunderstanding. But aren't we now only calling ```checkIfEdgeHeadingDifferenceExceedsThreshold(current, connection))``` between the starting edge and all of its connected edges? When we want to be calling it on every consecutive edge in the original Way. This might be doable if we reassign ```current``` whenever we move to the next layer of edges

Reply:
yes, thats true.
My assumption is when snakeRoadcheck is invoked its invoked for every edge, if we do this inside for loop, heading difference is calculated repeatedly for all consecutive edges.
So by not doing this only the edge where heading difference is greater than threshold is flagged.